### PR TITLE
BAU-Add-check-translation-to-yaml-precommit-hook

### PR DESCRIPTION
**Pull request by MattBidewell**
### What changed

When adding the translation files we should check to make sure that the keys are the same and throw warnings if they're not.

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: scripts/checkTranslation.js
Comment (MattBidewell) [inline scripts/checkTranslation.js:25]: to get this to work with nested files, extract out the for of loop above and do it recursively.

File: scripts/checkTranslation.js
Comment (MattBidewell) [inline scripts/checkTranslation.js:98]: We do it twice to compare once with the primary set being CY to EN then for EN to CY. To make sure no key are left behind.

File: src/locales/cy/default.yaml
Comment (MattBidewell) [inline src/locales/cy/default.yaml:1]: filler - translation will be in the next PR

File: package.json
Comment (MattBidewell) [inline package.json:78]: Dont merge until 0.0.23 is merged

File: .husky/pre-commit
Comment (MattBidewell) [inline .husky/pre-commit:5]: uncomment when welsh translations are in

**Review by merlinc — DISMISSED** [review]
Approving, as this LGTM once rebased

File: package.json
Comment (MattBidewell) [reply package.json:32]: We should leave this as tech debt because I'm sure a check translations script in an actual NPM DI package might prove useful for the whole of GDS

Comment (merlinc) [reply package.json:32]: :brain:

File: package.json
Comment (MattBidewell) [inline package.json:75]: 28 handles null values

**Review by merlinc — APPROVED** [review]

**Pull request merged**
